parseAPI/src/IA_power.C

```diff
@@ -316,12 +316,18 @@ bool IA_power::isReturn(Dyninst::ParseAPI::Function * context, Dyninst::ParseAPI
 	parsing_printf(" Not BLR - returning false \n");
 	return false;
    }
+
   Function *func = context;
-  parsing_printf
-    ("isblrReturn at 0x%lx Addr 0x%lx 0x%lx Function addr 0x%lx leaf %d \n",
-     current, getAddr (), currBlk->start (), context->addr (),
-     func->_is_leaf_function);
-  if (!func->_is_leaf_function)
+
+  parsing_printf("Checking return idioms for %s@0x%lx in block=0x%lx, context=0x%lx\n",
+                 context->name().c_str(), getAddr(), currBlk->start(), context->addr());
+
+  if(func->_is_leaf_function) {
```

Comment:
You added a check for the case of leaf function and return early, but it seems like you didnt remove the original code in the else clause of if(_is_leaf_function) that does the same thing?

And maybe we should stick to just one of func or context for consistency (even though they seem to have the same value as func is assigned in line 319).

Reply:
This is still a draft, so that code is actually going away. I wish GitHub would let you remove a review request, but it won't. Sorry.